src/twister2/device/hardware_adapter.py

```diff
@@ -54,17 +53,19 @@ def disconnect(self):
             self.connection.close()
             self.connection = None
             logger.info('Closed serial connection for %s', self.hardware_map.serial)
+        self.stop()
 
-    def flash(self, build_dir: str | Path, timeout: float = 60.0) -> None:
+    def run(self):
```

Comment:
Why such change? In future we want to have pytest-based tests which will allow to flash several independent images within a single test. Wouldn't this change make it harder, since it make a single image tightly coupled to the dut itself?

Reply:
The `DeviceAbstract` class inherits from `threading.Tread` class. `Thread` requires to implement `run` method, which is the target function started by thread. 
A usage of the code could be like so:
```python
t = DeviceAbstract()
t.start()  # start thread activity (`run`)
# run test code
t.join() # wait until the thread terminates.
```
but we need something like that:
```python
t = DeviceAbstract()
t.flash('build_dir')  # flash device and run
# run test code
t.stop() # stop running program
```
So in my implementation `flash` method just call `start` method, and `stop` call `join` method.

It should be possible to flash a device several times just like that:
```python
t = DeviceAbstract()
t.flash('build_dir_one')
# run test code
t.stop()
t.flash('build_dir_two')
# run test code
t.stop()
```